.github/workflows/test.yaml

```diff
@@ -35,6 +35,8 @@ jobs:
       env:
         POETRY_VERSION: "1.3.2"
       run: pip install poetry==${POETRY_VERSION} && poetry --version
+    - name: Check poetry.lock aligns with pyproject.toml
+      run: poetry lock --check
```

Comment:
Very nice! So this throws an error and causes this workflow to fail?  Can we test it in this branch somehow? 🤔

Reply:
It would be nice to be able to test this, it should cause an error though. I went under the assumption that the CI uses the exit code of a command to determine success or failure. When we run `poetry lock --check` and the lock file is out of sync the exit code, `echo $?`, returns 1, which maps to `Operation not permitted`. It's the same for test failures with `make test`.

Reply:
That's a good idea! I'll try it out.

Reply:
[Looks like it does the trick!](https://github.com/cds-snc/notification-api/actions/runs/4995443383/jobs/8947313443?pr=1866)

Reply:
Nice!